common/src/main/java/com/microsoft/identity/common/internal/cache/MsalCppOAuth2TokenCache.java

```diff
@@ -93,37 +91,26 @@ public static MsalCppOAuth2TokenCache create(@NonNull final Context context) {
     }
 
     /**
-     * @param accountRecord : AccountRecord associated with the input credentials.
+     * @param accountRecord : AccountRecord associated with the input credentials, can be null.
      * @param credentials   : list of Credential which can include AccessTokenRecord, IdTokenRecord and RefreshTokenRecord.
-     *                      Note : Both IdTokenRecord and RefreshTokenRecord need to be non null. AccessTokenRecord can be optional.
      * @throws ClientException : If the supplied Account or Credential are null or schema invalid.
      */
-    public synchronized void saveCredentials(@NonNull final AccountRecord accountRecord,
+    public synchronized void saveCredentials(@Nullable final AccountRecord accountRecord,
                                              @NonNull final Credential... credentials) throws ClientException {
         if (credentials == null || credentials.length == 0) {
             throw new ClientException("Credential array passed in is null or empty");
         }
 
-        AccessTokenRecord accessTokenRecord = null;
-        IdTokenRecord idTokenRecord = null;
         RefreshTokenRecord refreshTokenRecord = null;
 
         for (final Credential credential : credentials) {
-            if (credential instanceof AccessTokenRecord) {
-                accessTokenRecord = (AccessTokenRecord) credential;
-            } else if (credential instanceof IdTokenRecord) {
-                idTokenRecord = (IdTokenRecord) credential;
-            } else if (credential instanceof RefreshTokenRecord) {
+            if (credential instanceof RefreshTokenRecord) {
                 refreshTokenRecord = (RefreshTokenRecord) credential;
             }
         }
-
-        validateNonNull(accountRecord, "AccountRecord");
-        validateNonNull(refreshTokenRecord, "RefreshTokenRecord");
-        validateNonNull(idTokenRecord, "IdTokenRecord");
-        validateCacheArtifacts(accountRecord, accessTokenRecord, refreshTokenRecord, idTokenRecord);
-
-        removeRefreshTokenIfNeeded(accountRecord, refreshTokenRecord);
+        if (accountRecord != null && refreshTokenRecord != null) {
+            removeRefreshTokenIfNeeded(accountRecord, refreshTokenRecord);
+        }
```

Comment:
Just for my own edification, can you add a comment here?
```suggestion
        if (accountRecord != null && refreshTokenRecord != null) {
            // This will not be true when we are called from MSAL C++, as the account will always be null.
            removeRefreshTokenIfNeeded(accountRecord, refreshTokenRecord);
        }
```
Feel free to reword.